contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -335,6 +351,68 @@ func (h *htlcIncomingContestResolver) report() *ContractReport {
 	}
 }
 
+// handOffHtlcClaim returns the inner htlcSuccessResolver so that we can claim
+// the incoming htlc on chain with the preimage. If the htlc is on our
+// commitment and we need to claim the htlc in two stages, we record stage one
+// of the htlc claim here.
+func (h *htlcIncomingContestResolver) handOffHtlcClaim() (ContractResolver,
+	error) {
+
+	// If we do not have a signed success transaction, we do not need to
+	// record the first stage of the two step htlc claim process, because
+	// this htlc will simply be claimed directly from the remote party's
+	// commitment.
+	if h.htlcResolution.SignedSuccessTx == nil {
+		return &h.htlcSuccessResolver, nil
+	}
+
+	// If the SignedSuccessTx is not nil, we are claiming the htlc in two
+	// stages from our own commitment. Here, we record the intermediate
+	// SignedSuccessTx stage so that we have a complete record of all on
+	// chain transactions that were required to claim this htlc.
+	spendTx := h.htlcResolution.SignedSuccessTx
+	spendTxID := spendTx.TxHash()
+
+	// To get the amount of the sole input to our signed success tx, we
+	// lookup the input on our channel's commitment to get its value. We can
+	// look this outpoint up because the success tx spends from our commit.
+	outpoint := spendTx.TxIn[0].PreviousOutPoint
+
+	output, err := h.FindOwnOutput(&outpoint)
+	if err != nil {
+		return nil, err
+	}
+	amt := btcutil.Amount(output.Value)
```

Comment:
If this is just to get the amount for the HTLC itself, then this is already available in the internal `htlc` field which contains the full amount of the HTLC. 

The other way this information is available is from the internal `htlcResolution` struct. The sign desc contains the input value since we need that to generate a valid signature. So the path here is `h.htlcResolution.SweepSignDesc.Output.Value`.